habitat-lab/habitat/isaacsim/usd_conversion_scene.py

```diff
@@ -0,0 +1,296 @@
+
+    # Convert .collision.glb to USD with isaacsim omniverse
+    # Output .usd file in the object folder that contains collision.glb
+
+    # CONVERSION
+
+    # Turn object .glb to xform
+
+    # Add count to string if object used more than once
+    if count > 0:
+        object_name = object_name + f"_{count}"
+
+    object_stage_name = "/OBJECT_" + object_name # NOTE: Object state name must start with a char letter
+
+    object_xform = UsdGeom.Xform.Define(
+        stage, object_stage_name
+    )  
+    object_xform.GetPrim().GetReferences().AddReference(object_filepath)
+
+    # Set Position
+    translate_op = next(
+        (
+            op
+            for op in object_xform.GetOrderedXformOps()
+            if op.GetName() == "xformOp:translate"
+        ),
+        None,
+    )
+    if translate_op is None:
+        translate_op = object_xform.AddTranslateOp()
+    translate_op.Set(value=Gf.Vec3d(*position))
+
+    # Set Rotation
+    orient_op = next(
+        (
+            op
+            for op in object_xform.GetOrderedXformOps()
+            if op.GetName() == "xformOp:orient"
+        ),
+        None,
+    )
+    if orient_op is None:
+        orient_op = object_xform.AddOrientOp()
+    orient_op.Set(value=Gf.Quatd(*rotation))
+
+    # Set Scale
+    scale_op = next(
+        (
+            op
+            for op in object_xform.GetOrderedXformOps()
+            if op.GetName() == "xformOp:scale"
+        ),
+        None,
+    )
+    if scale_op is None:
+        scale_op = object_xform.AddScaleOp()
+    scale_op.Set(value=Gf.Vec3d(*scale))
+
+
+def convert_scene_instance(
+    scene_instance_filepath: str, stage_glb_folder: str, local_download_folder: str, glb_data_folder: str
+) -> None:
+    """
+    This function takes a folder of .glb objects, converts them remotely to .usd,
+    downloads the converted files, and then makes a .usda scene. The idea is to
+    convert a scene_instance json to a .usd file by aggregating all the defined 
+    objects within the scene instance json.
+    """
+    
+    # Open scnece_instance JSON
+    with open(scene_instance_filepath, "r") as file:
+        scene_data = json.load(file)
+ 
+    
+    stage_name: str = scene_data[STAGE_INSTANCE][TEMPLATE_NAME].split('/')[1
+    ]
+    
+    # Form a list of existing object glb file paths 
+    object_glb_filepath_list = make_object_filepath_list(
+        glb_data_folder,
+        scene_instance_filepath,
+        scene_data,
+        file_extension=".collider.glb",
+    )
+
+    # # Append stage glb to piggy back on conversion.py. #Note, stage glb not necessary probably
+    #stage_glb_filepath = stage_glb_folder + stage_name + '.glb'
+    #object_glb_filepath_list.append(stage_glb_filepath)
+
+    # Upload glb objects to remote server
+    num_processes = 10
+    barrier = Barrier(num_processes)
+    with Pool(processes=num_processes) as pool:
+        pool.map(upload_object_glb_to_server, object_glb_filepath_list)
```

Comment:
Let's not use SSH for anything in these scene-conversion scripts. I understand you are using SSH for your development, but in practice, this script is going to run locally on a machine on which HSSD and other assets are present, and the converted USD files should be written to the local filesystem.

I suspect the reason you took this approach is that you didn't have a practical way to upload HSSD to my Lambda machine? I have it on that machine, just not under the guest account. I can add a copy to the guest account so you can do all your conversion testing remotely on the Lambda machine, without having to upload or download any files to/from your local machine. Let's chat more if you're blocked here!